File: easy/Black-Card/Black Card.cs
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            BlackDot(line);
        }
    }
    
    static void BlackDot(string line){
        int pos = line.IndexOf('|');
        string[] playersStr = line.Substring(0,pos).Trim().Split(' ');
        int dot = Convert.ToInt32(line.Substring(pos + 1).Trim());
        ArrayList players = new ArrayList();
        players.AddRange(playersStr);
        while(players.Count>1){
            int dotCounter = 1;
            int index = 0;
            while(true){
                if(dot!=dotCounter){
                    index++;
                    dotCounter++;
                    if(index==players.Count)index=0;
                }else {
                    players.RemoveAt(index);
                    break;
                }
            }
        }
        Console.WriteLine(players[0]);
    }
}

File: easy/Black-Card/Black Card.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            BlackDot(line);
        }
    }
    
    static void BlackDot(String line){
        int pos = line.indexOf("|");
        String[] playersStr = line.substring(0,pos).trim().split("\\s+");
        int dot = Integer.valueOf(line.substring(pos + 1).trim());
        ArrayList<String> players = new ArrayList(Arrays.asList(playersStr));
        while(players.size()>1){
            int dotCounter = 1;
            int index = 0;
            while(true){
                if(dot!=dotCounter){
                    index++;
                    dotCounter++;
                    if(index==players.size())index=0;
                }else {
                    players.remove(index);
                    break;
                }
            }
        }
        System.out.println(players.get(0));
    }
}
